well_formedness.rb

```diff
@@ -0,0 +1,46 @@
+require 'pry'
+
+class Validator
+  def validate(string)
+    result = get_relevant_chars(string)
+    return false if result.length.odd?
+    check_chars_match(result)
+
+
+  end
+
+  def get_relevant_chars(string)
+    result = []
+    string.each_char do |char|
```

Comment:
Which do you think is more efficient, using `each_char` with a collector local variable, or splitting the string and then mapping over it?